classes/migration/SubmissionFilesMigration.inc.php

```diff
@@ -24,63 +23,44 @@ class SubmissionFilesMigration extends Migration {
         public function up() {
 		// Files associated with submission. Includes submission files, etc.
 		Capsule::schema()->create('submission_files', function (Blueprint $table) {
-			$table->bigInteger('file_id')->autoIncrement();
-			$table->bigInteger('revision');
-			$table->bigInteger('source_file_id')->nullable();
-			$table->bigInteger('source_revision')->nullable();
+			$table->bigInteger('submission_file_id')->autoIncrement();
 			$table->bigInteger('submission_id');
-			$table->string('file_type', 255);
+			$table->bigInteger('file_id');
+			$table->bigInteger('source_submission_file_id')->nullable();
 			$table->bigInteger('genre_id')->nullable();
-			$table->bigInteger('file_size');
-			$table->string('original_file_name', 127)->nullable();
 			$table->bigInteger('file_stage');
 			$table->string('direct_sales_price', 255)->nullable();
 			$table->string('sales_type', 255)->nullable();
 			$table->smallInteger('viewable')->nullable();
-			$table->datetime('date_uploaded');
-			$table->datetime('date_modified');
+			$table->datetime('created_at');
```

Comment:
:+1: for starting to introduce Laravel conventions here. Looking at https://laravel.com/docs/7.x/migrations#creating-tables you could use a one-liner to add both datetimes, I think...
````
$table->timestamps(0);
```

Reply:
It looks like that adds `TIMESTAMP` columns but these columns are `DATETIME`. Are these equivalent?

Reply:
Essentially, `TIMESTAMP` columns don't do dates before 1970 in MySQL. I think that'll be fine.

Reply:
I'm running into fatal errors when trying to do update to a `TIMESTAMP` column. The installation works fine, but the migration does not. It looks like this may not be supported by Doctrine:

https://github.com/doctrine/dbal/issues/2558

This is the code I was trying on the migration:

```
		Capsule::schema()->table('submission_files', function (Blueprint $table) {
			$table->renameColumn('file_id', 'submission_file_id');
			$table->renameColumn('new_file_id', 'file_id');
			$table->renameColumn('source_file_id', 'source_submission_file_id');
			$table->renameColumn('date_uploaded', 'created_at');
			$table->renameColumn('date_modified', 'updated_at');
			$table->timestamp('created_at', 0)->change();
			$table->timestamp('updated_at', 0)->change();
			$table->dropColumn('revision');
			$table->dropColumn('source_revision');
			$table->dropColumn('file_size');
			$table->dropColumn('file_type');
			$table->dropColumn('original_file_name');
			$table->foreign('file_id')->references('file_id')->on('files');
		});
```

And the error I received:

```
RROR: Upgrade failed: DB: Unknown column type "timestamp" requested. Any Doctrine type that you use has to be registered with \Doctrine\DBAL\Types\Type::addType(). You can get a list of all the known types with \Doctrine\DBAL\Types\Type::getTypesMap(). If this error occurs during database introspection then you might have forgotten to register all database types for a Doctrine Type. Use AbstractPlatform#registerDoctrineTypeMapping() or have your custom types implement Type#getMappedDatabaseTypes(). If the type name is empty you might have a problem with the cache or forgot some mapping information.
```

Thoughts?

Reply:
OK, let's stick with datetime until we have a compelling reason to dig into this further. Probably datestamps are intended for automatic management (i.e. populate on create) and our current uses might fall outside of that.